src/mesh/impls/bout/boutmesh.cxx

```diff
@@ -1806,12 +2013,33 @@ void BoutMesh::set_connection(int ypos1, int ypos2, int xge, int xlt, bool ts) {
         UDATA_INDEST = UDATA_OUTDEST;
       UDATA_XSPLIT = xge;
       UDATA_OUTDEST = PROC_NUM(PE_XIND, ypedown);
-      if (UDATA_XSPLIT <= 0)
+      if (UDATA_XSPLIT <= 0) {
         UDATA_INDEST = -1;
+      }
 
       TS_up_out = ts;
       output_info.write("=> This processor sending out up\n");
     }
+
+    if (include_corner_cells) {
+      // Connect corners as if communicating all guard cells first in y, then in x
```

Comment:
I fail to see why you not do that?
That would keep the number of communications the same, only introduce slightly larger messages.

What you would do is first communicate in y as we do now. Then you communicate in x, but including the boundary cells. This way you need the name number of communications, which would mean less impact on the runtime, as the package size change only increases slightly.

Reply:
The trouble is you'd have to wait for the y-communication to complete before you can do the x-communication; at the moment, and in the changes proposed here, all the communications can be sent at the same time, and you only have to call `wait` once. So while it's true that the amount of information passed would not change much, I'm under the impression that communications are limited by latency rather than bandwidth, so we don't want to wait for two sets of communication to complete rather than one.

Reply:
I guess we should do some proper benchmarking ...

A simple way to do this, without any code changes should be switching
between async and sync send, where we already have a switch for.
If we are limited by latencies, the async version should make the
communication roughly 4x as fast, but as far as I remember this is not
the case, but I might be wrong.

It's been a while, but I seem to remember that the async switch didn't
have a significant impact.

Reply:
The switch is `async_send`, which using `MPI_Isend` instead of `MPI_Send`, but because MPI can copy the buffer and return from `MPI_Send` without needing to actually complete the communication, this may not make much difference https://stackoverflow.com/questions/10017301/mpi-blocking-vs-non-blocking. I just did a quick test and communicating over 4 nodes of Marconi, `async_send` made no noticable difference. However, the receives in BOUT++ are always asynchronous - I think that means that regardless of the `async_send` setting, the communications are overlapping. It would need more hacking around to force the x- and y-communications to really be consecutive to test this further - would be interesting to prove there's a difference, but I'm not going to pursue now.